### DISCUSSION TITLE
上架Microsoft Store的计划

### DISCUSSION THREAD

**Discussion by StarsbySea** [Richasy/Bili.Uwp#340] [常规]
众所周知，Windows内的根证书存储几乎被所有软件信任（如Chrome），添加根证书会给系统增加安全隐患。
上架微软商店好处众多，比如在商店内下载更新，也可以自己分发安装包（经过微软签名的安装包可以直接安装），缺点就是注册成为微软开发者需要一定费用（可考虑开放捐赠，但可能有法律风险），而且可能不能提供一些会侵犯版权的功能。
关于通过自己分发应用，可以参考[Unigram](https://github.com/UnigramDev/Unigram)的方法
不知道开发者对于上架商店有无打算，有何计划呢？

**Comment by tm0210** [comment]
上架了，然后被一封律师函给下架了😓

**Reply by tm0210** [reply]
所以，别问，问就是律师函警告⚠️

**Reply by tm0210** [reply]
![IMG_20210929_162120](https://user-images.githubusercontent.com/90995212/135230819-b674c093-e806-4918-9252-5f62984892a5.jpg)

**Comment by Richasy** [comment]
是的，事实上已经上架过，但为了避免这种法律问题，已经下架了

**Reply by run-ze** [reply]
> 是的，事实上已经上架过，但为了避免这种法律问题，已经下架了

好可惜，如果上架，还可以期望在xbox上使用。官方版操作别扭而且容易闪退。

**Reply by run-ze** [reply]
只上外区可以规避这种问题吗？

**Reply by Richasy** [reply]
> 只上外区可以规避这种问题吗？

我不太想给自己惹麻烦，这对我来说没必要

**Reply by ghost** [reply]
还有个曲线救国的手法, 上架winget..
(虽然这依然会带来麻烦, 但考虑到会用命令行的人数, 这已经和目前的安装方式相差无几了

**Comment by TigerCubDen** [comment]
开发者是否允许第三方开发者/用户打包本项目上架到微软商店？条件为保持免费和不添加任何私货。

**Reply by Richasy** [reply]
根据MIT开源协议，我不会干涉使用者对哔哩项目的二次分发，只要按照开源协议的要求进行署名和导引即可。

但从个人角度，我依然不建议这种行为。虽然有开源协议，但权责很难划分清楚，打包的项目一旦被举报，直接责任人是谁？ 我是否有权要求打包作品下架，产生的直接或间接收益如何处理等问题都会很棘手

**Comment by cnoim** [comment]
好像已经拥有的用户在正式版出来之后可以获取更新吧？

**Reply by AkazaRenn** [reply]
是可以，甚至可以通過安裝 msixbundle 強行「擁有」。但是更新還是得開發者主動在後臺更新過才行，現在商店裡的還是去年的版本。

**Comment by AkazaRenn** [comment]
不知道可不可以在下架的 Microsoft Store 後臺更新成最新的版本，這樣大家可以靠安裝老的 msixbundle 來獲得自動更新

**Reply by Sunovn** [reply]
就像AppStore一样吗

**Reply by AkazaRenn** [reply]
是的，因為 Windows 可以 sideload 所以可以做這種騷操作

**Comment by mnixry** [comment]
问一下，是否可以考虑添加到`winget`或者`scoop`这种命令行包管理器中

**Comment by StarsbySea** [comment]
该App已经实质上可以从商店安装，本贴完结。具体参考README.md文件
